custom_components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,448 @@
+        if self._preset_list and PRESET_AWAY in self._preset_list:
+            support |= SUPPORT_TARGET_TEMPERATURE_RANGE
+        if self.values.fan_mode:
+            support |= SUPPORT_FAN_MODE
+        if self._preset_list:
+            support |= SUPPORT_PRESET_MODE
+        return support
+
+    def _update_operation_mode(self):
+        """Update hvac and preset modes."""
+        if not self._mode():
+            return
+        self._hvac_list = []
+        self._preset_list = []
+        self._hvac_value_label_mapping = {}
+        self._hvac_label_value_mapping = {}
+        values_list = self._mode().value.get(VALUE_LIST)
+        mode_values = list(map(lambda x: x.get(VALUE_ID), values_list))
+        for value in mode_values:
+            ha_mode = HVAC_MODE_MAPPINGS.get(value)
+            if ha_mode is not None and ha_mode not in self._hvac_list:
+                self._hvac_list.append(ha_mode)
+            else:
+                self._preset_list.append(value)
+            label = next(
+                (
+                    entry.get(VALUE_LABEL)
+                    for entry in values_list
+                    if value == entry.get(VALUE_ID)
+                )
+            )
+            self._hvac_value_label_mapping[value] = label
+            self._hvac_label_value_mapping[label.lower()] = value
+
+        for mode in DEFAULT_HVAC_MODES:
+            if mode in self._hvac_list:
+                self._default_hvac_mode = mode
+                break
+
+        current_mode_value = self._mode().value.get(VALUE_SELECTED)
+        if current_mode_value in HVAC_MODE_MAPPINGS:
+            self._hvac_mode = current_mode_value
+            self._preset_mode = PRESET_NONE
+        else:
+            current_mode_label = self._hvac_value_label_mapping.get(current_mode_value)
+            if (
+                "heat" in current_mode_label.lower()
+                and HVAC_MODE_HEAT in self._hvac_list
+            ):
+                self._hvac_mode = 0x01
+            elif (
+                "cool" in current_mode_label.lower()
+                and HVAC_MODE_COOL in self._hvac_list
+            ):
+                self._hvac_mode = 0x02
+            else:
+                self._hvac_mode = self._default_hvac_mode
+            self._preset_mode = current_mode_value
+
+    def _update_current_temp(self):
+        """Update current temperature."""
+        if not self.values.temperature:
+            return
+        self._current_temperature = self.values.temperature.value
+        device_unit = self.values.temperature.units
+        if device_unit is not None:
+            self._unit = device_unit
+
+    def _update_fan_mode(self):
+        """Update fan mode."""
+        if not self.values.fan_mode:
+            return
+        self._fan_value_label_mapping = {}
+        self._fan_label_value_mapping = {}
+        self._fan_modes = []
+        for entry in self.values.fan_mode.value.get(VALUE_LIST):
```

Comment:
Please use `dict[key]` when we know that the key is in the dict.
```suggestion
        for entry in self.values.fan_mode.value[VALUE_LIST]:
```